src/overlays/actors/ovl_En_Fall/z_en_fall.c

```diff
@@ -36,75 +64,820 @@ const ActorInit En_Fall_InitVars = {
+    }
+}
+
+void EnFall_Moon_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    // This offsets the moon's focus so that the Moon's Tear actually falls
+    // out of its eye when looking at it through the telescope.
+    static Vec3f sFocusOffset[] = { 1800.0f, 1000.0f, 4250.0f };
+    EnFall* this = THIS;
+    s32 primColor;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    Matrix_MultiplyVector3fByState(sFocusOffset, &this->actor.focus.pos);
+    primColor = (this->eyeGlowIntensity * 200.0f) + 40.0f;
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0x80, primColor, primColor, primColor, 255);
+    gSPDisplayList(POLY_OPA_DISP++, D_060077F0);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void EnFall_OpenMouthMoon_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnFall* this = THIS;
+    s32 primColor;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    primColor = (this->eyeGlowIntensity * 200.0f) + 40.0f;
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0x80, primColor, primColor, primColor, 255);
+    gSPDisplayList(POLY_OPA_DISP++, D_06002970);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void EnFall_LodMoon_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnFall* this = THIS;
+    s32 primColor;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    POLY_OPA_DISP = Gfx_SetFog(POLY_OPA_DISP, 20, 25, 30, 0, 0x3E7, 0x3200);
+    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_PASS, G_RM_AA_ZB_OPA_SURF2);
+    gSPLoadGeometryMode(POLY_OPA_DISP++, G_ZBUFFER | G_SHADE | G_CULL_BACK | G_SHADING_SMOOTH);
+    primColor = (this->eyeGlowIntensity * 200.0f) + 40.0f;
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0x80, primColor, primColor, primColor, 255);
+    gSPDisplayList(POLY_OPA_DISP++, D_060010E0);
+    gSPLoadGeometryMode(POLY_OPA_DISP++, G_ZBUFFER | G_SHADE | G_CULL_BACK | G_LIGHTING | G_SHADING_SMOOTH);
+    gSPDisplayList(POLY_OPA_DISP++, D_06001158);
+    POLY_OPA_DISP = func_801660B8(globalCtx, POLY_OPA_DISP);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void EnFall_LodMoon_DrawWithoutLerp(Actor* thisx, GlobalContext* globalCtx) {
+    EnFall_LodMoon_Draw(thisx, globalCtx);
+}
+
+/**
+ * If the moon is more than 9000 units away from the eye, this will lerp it
+ * to be 9000 units away before drawing it.
+ */
+void EnFall_LodMoon_DrawWithLerp(Actor* thisx, GlobalContext* globalCtx) {
+    f32 distanceToEye;
+    f32 scale;
+    Vec3f translation;
+
+    distanceToEye = Actor_DistanceToPoint(thisx, &globalCtx->view.eye);
+    if (distanceToEye > 9000.0f) {
```

Comment:
```suggestion
    f32 distanceToEye = Actor_DistanceToPoint(thisx, &globalCtx->view.eye);
    f32 scale;
    Vec3f translation;

    if (distanceToEye > 9000.0f) {
```
I think in this case, it'd be fine if you wanted to leave it as is. Up to you